components/img_generator/img_generator.py

```diff
@@ -0,0 +1,167 @@
+from PIL import Image, ImageDraw, ImageSequence, ImageFont
+from protos.activity_id_pb2 import ActivityID
+from protos.rated_player_pb2 import RatedPlayer
+
+# Coordinates relative to (X=0, Y=0) for any section on the GIF.
+# Name is short to keep the logic compact in methods below.
+COORDS = {
+    "activity": (20, 20),
+    "details": (20, 40),
+    "players": [
+        (30, 60),
+        (30, 85),
+        (30, 110),
+        (30, 135),
+        (30, 160),
+        (30, 185)
+    ],
+    "substitutes": [
+        (220, 148),
+        (220, 168),
+        (220, 188)
+    ],
+    "section_w": 350,
+    "delta_x": 486,
+    "delta_y": 260,
+}
+
+ACTIVITY_NAMES = {
+    ActivityID.Type.LEVIATHAN: "Léviathan",
+    ActivityID.Type.EATER_OF_WORLDS: "Dévoreur de Mondes",
+    ActivityID.Type.SPIRE_OF_STARS: "Flèche d'étoiles",
+    ActivityID.Type.CROWN_OF_SORROW: "Couronne du Malheur",
+    ActivityID.Type.LAST_WISH: "Dernier Voeu",
+    ActivityID.Type.SCOURGE_OF_THE_PAST: "Fléau du Passé",
+    ActivityID.Type.GARDEN_OF_SALVATION: "Jardin du Salut",
+    ActivityID.Type.VAULT_OF_GLASS: "Caveau de Verre",
+    ActivityID.Type.CROPTAS_END: "Cropta",
+    ActivityID.Type.THE_TAKEN_KING: "La Chute du Roi",
+    ActivityID.Type.WRATH_OF_THE_MACHINE: "Fureur Mécanique",
+}
+
+# Add Prestige names when needed
+for activity_id in list(ACTIVITY_NAMES):
+    name_prestige = ActivityID.Type.Name(activity_id) + "_PRESTIGE"
+    if name_prestige in ActivityID.Type.keys():
+        prestige_id = ActivityID.Type.Value(name_prestige)
+        ACTIVITY_NAMES[prestige_id] = ACTIVITY_NAMES[activity_id] + " - Prestige"
+
+CR = ImageFont.truetype('components/img_generator/assets/calibri_bold.ttf', 16)
+CI = ImageFont.truetype('components/img_generator/assets/calibri_italic.ttf', 16)
+
+COLORS = {
+    "blue": (78, 96, 224, 255),
+    "purple": (144, 9, 226, 255),
+    "black": (0, 0, 0, 255),
+    "green": (15, 155, 73, 255),
+    "orange": (188, 96, 15, 255),
+}
+
+COLORS_BY_RATING = {
+    RatedPlayer.Rating.UNKNOWN: COLORS["black"],
+    RatedPlayer.Rating.BEGINNER: COLORS["green"],
+    RatedPlayer.Rating.INTERMEDIATE: COLORS["orange"],
+    RatedPlayer.Rating.EXPERIENCED: COLORS["blue"],
+}
+
+NUM_SECTIONS = 4
+
+
+class Generator:
+    """Generates the planning in GIF format."""
+
+    def __init__(self, planning, date_tz):
+        self.__planning = planning
+        self.__date_tz = date_tz
+
+    def move(self, coordinates, section, width_to_center=None):
+        """
+        Moves coordinates at the right place depending on the section and the center preferences.
+        :param coordinates: tuple (X,Y) representing the initial coordinates.
+        :param section: the section index, e.g. from 0 to 3 for the 1st GIF.
+        :param width_to_center: optional int indicating the width of text if it has to be centered.
+        :return: tuple for the new coordinates.
+        """
+        dx = COORDS["delta_x"] if (section % 2 == 1) else 0
```

Comment:
How about a number of rows and columns constant? :P

Reply:
Pushing back on this for now, would change a good deal of the coordinates logic and we have only seen 2x2 banners so far ;)
Can be a feature when we have a V0 up and running